presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveFileSystemS3SelectPushdown.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.s3.HiveS3Config;
+import com.facebook.presto.hive.s3.PrestoS3ConfigurationUpdater;
+import com.facebook.presto.hive.s3.S3ConfigurationUpdater;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static org.testng.util.Strings.isNullOrEmpty;
+
+public class TestHiveFileSystemS3SelectPushdown
+        extends AbstractTestHiveFileSystem
+{
+    private String awsAccessKey;
+    private String awsSecretKey;
+    private String writableBucket;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket"})
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket)
+    {
+        checkArgument(!isNullOrEmpty(host), "Expected non empty host");
+        checkArgument(!isNullOrEmpty(databaseName), "Expected non empty databaseName");
+        checkArgument(!isNullOrEmpty(awsAccessKey), "Expected non empty awsAccessKey");
+        checkArgument(!isNullOrEmpty(awsSecretKey), "Expected non empty awsSecretKey");
+        checkArgument(!isNullOrEmpty(writableBucket), "Expected non empty writableBucket");
+
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+
+        super.setup(host, port, databaseName, this::createHdfsConfiguration, true);
+    }
+
+    private HdfsConfiguration createHdfsConfiguration(HiveClientConfig config)
```

Comment:
is this identical to `com.facebook.presto.hive.TestHiveFileSystemS3#createHdfsConfiguration`?

Reply:
Yes. I was not sure about the intent future intent of AbstractTestHiveFileSystem. Thus I didn't push the method in the abstract class. Would you prefer us to move the method over there?

Reply:
I would introduce `AbstractTestHiveFileSystemS3` to hold functionality common to `TestHiveFileSystemS3` and your new class, and made `TestHiveFileSystemS3` and your new class extend `AbstractTestHiveFileSystemS3`.

Reply:
Makes sense.